src/diffpy/labpdfproc/tools.py

```diff
@@ -51,16 +58,21 @@ def set_input_lists(args):
         input_path = Path(input).resolve()
         if input_path.exists():
             if input_path.is_file():
-                input_paths.append(input_path)
+                if "file_list" in input_path.name:
```

Comment:
I think the code will be cleaner and easier to read if we just do a prefilter, so we don't change any code down here but we change the args.input up above to "expand" file_list into its list of files.

It could maybe even not be a private function but a public function that is called before `set_input_file_list`.  We could call it `expand_list_file` and it just operates on strings.

Reply:
If the objective is to handle duplicated files, would it be easier to use set(input_paths) at the end, so no other function is needed?

Reply:
no, it is just keeping the code easier to read, so it doesn't get like spaghetti with lots of complicated conditionals handling edge cases all the way through it.  it is about maintainability in the future.  None of it matters that much, but it is also a kind of learning opportunity where we develop better practices in the group as a whole.

I think we will find if we do it this way the code will be easier to follow and the different tasks will be handled in a nicely separated way.  For example we could call this function or not call it and the code would still work either way, but by calling it we get new functionality allowing the user to input lists of files from a text file.